starknet_py/utils/typed_data.py

```diff
@@ -242,58 +283,114 @@ def _verify_types(self):
             if type_name in self.types:
                 raise ValueError(f"Reserved type name: {type_name}")
 
-        referenced_types = {
-            ref_type.contains
-            if ref_type.contains is not None
-            else strip_pointer(ref_type.type)
-            for type_name in self.types
-            for ref_type in self.types[type_name]
-        }
+        referenced_types = set()
+        for type_name in self.types:
+            for ref_type in self.types[type_name]:
+                if isinstance(ref_type, MerkleTreeParameter):
+                    referenced_types.add(ref_type.contains)
+                elif isinstance(ref_type, EnumParameter):
+                    self._validate_enum_type()
```

Comment:
Not sure about the validation here. What about other basic types from revision 1 that are not validated this way? Why is it only for `Enum`?

Reply:
This validation is performed, because a custom enum type **could not** be defined in revision 0. For example, `timestamp` is defined in rev 1, so if user defined it in revision 0, it would be simply treated as a custom type. However, as per [type identification](https://github.com/starknet-io/SNIPs/pull/77/files#diff-d594c362a3c260ea39dacaca55a6f02927013cebd19a5c8d202117ff9147ff14R124), this does not apply to enums.

Reply:
Thanks, as discussed offline let's leave it as it is.